app/src/main/java/de/danoeh/antennapod/fragment/InboxFragment.java

```diff
@@ -63,19 +68,12 @@ public boolean onMenuItemClick(MenuItem item) {
             return true;
         }
         if (item.getItemId() == R.id.remove_all_inbox_item) {
-            ConfirmationDialog removeAllNewFlagsConfirmationDialog = new ConfirmationDialog(getActivity(),
-                    R.string.remove_all_inbox_label,
-                    R.string.remove_all_inbox_confirmation_msg) {
-
-                @Override
-                public void onConfirmButtonPressed(DialogInterface dialog) {
-                    dialog.dismiss();
-                    DBWriter.removeAllNewFlags();
-                    ((MainActivity) getActivity()).showSnackbarAbovePlayer(
-                            R.string.removed_all_inbox_msg, Toast.LENGTH_SHORT);
-                }
-            };
-            removeAllNewFlagsConfirmationDialog.createNewDialog().show();
+            shouldNotPromptRemoveAll = UserPreferences.shouldNotPromptRemoveAllFromInboxDialog();
+            if (shouldNotPromptRemoveAll) {
```

Comment:
Why does this need to be stored in a member variable? You can directly use the value from the preferences here